weatherman.py

```diff
@@ -0,0 +1,246 @@
+__weather__ = {}  # for storing weather records
+__months__ = {1: "Jan", 2: "Feb", 3: "Mar", 4: "Apr", 5: "May", 6: "Jun", 7: "Jul", 8: "Aug", 9: "Sep", 10: "Oct",
+              11: "Nov", 12: "Dec"}
+
+import os
+import sys
+
+
+class Parser:
```

Comment:
Parser and Reader are two different things! We are achieving both of the things via single class and calling it Parser. Split both to dedicated classes.